edgelet/edgelet-http-workload/src/server/decrypt.rs

```diff
@@ -48,255 +53,57 @@ where
-            .unwrap();
-
-        assert_eq!(b64_reversed!(), body.plaintext().to_string());
-    }
-
-    #[test]
-    fn handler_responds_with_bad_request_when_params_are_missing() {
-        let (request, params) = args_with_empty_params();
-        let handler = DecryptHandler::new(TestHsm::default());
-
-        let response = handler.handle(request, params).wait().unwrap();
-
-        assert_eq!(StatusCode::BAD_REQUEST, response.status());
-        assert_response_message_eq("The request is missing required parameter `name`", response);
-    }
-
-    #[test]
-    fn handler_responds_with_bad_request_when_name_is_missing() {
-        let (request, params) = args_with_no_name();
-        let handler = DecryptHandler::new(TestHsm::default());
-
-        let response = handler.handle(request, params).wait().unwrap();
-
-        assert_eq!(StatusCode::BAD_REQUEST, response.status());
-        assert_response_message_eq("The request is missing required parameter `name`", response);
-    }
-
-    #[test]
-    fn handler_responds_with_bad_request_when_genid_is_missing() {
-        let (request, params) = args_with_no_genid();
-        let handler = DecryptHandler::new(TestHsm::default());
-
-        let response = handler.handle(request, params).wait().unwrap();
-
-        assert_eq!(StatusCode::BAD_REQUEST, response.status());
-        assert_response_message_eq(
-            "The request is missing required parameter `genid`",
-            response,
-        );
-    }
-
-    #[test]
-    fn handler_responds_with_bad_request_when_request_is_malformed() {
-        let (request, params) = args_with_bad_request();
-        let handler = DecryptHandler::new(TestHsm::default());
-
-        let response = handler.handle(request, params).wait().unwrap();
-
-        assert_eq!(StatusCode::BAD_REQUEST, response.status());
-        assert_response_message_eq(
-            "Request body is malformed\n\tcaused by: expected value at line 1 column 1",
-            response,
-        );
-    }
-
-    #[test]
-    fn handler_responds_with_unprocessable_entity_when_request_args_are_not_base64_encoded() {
-        let bodies = &[
-            request_with_unencoded_ciphertext(),
-            request_with_unencoded_init_vector(),
-        ];
-        let handler = DecryptHandler::new(TestHsm::default());
-
-        for body in bodies {
-            let (request, params) = create_args(Some(body), params_ok!());
-            let response = handler.handle(request, params).wait().unwrap();
-
-            assert_eq!(StatusCode::BAD_REQUEST, response.status());
-            assert_response_message_eq(
-                "Request body is malformed\n\tcaused by: Encoded text cannot have a 6-bit remainder.",
-                response,
-            );
-        }
-    }
+fn get_plaintext(key_client: Arc<Mutex<aziot_key_client::Client>>, key_handle: KeyHandle, iv: Vec<u8>, ciphertext: Vec<u8>) -> impl Future<Item = Vec<u8>, Error = Error> {
+	let aad = b"$iotedged".to_vec();
```

Comment:
The AAD is the module metadata. In the old code it was `format!("{}{}", module_id, module_generation_id).to_vec()`

Reply:
Fixed